R/dashboard.R

```diff
@@ -0,0 +1,134 @@
+#' R Markdown format for model monitoring dashboards
+#'
+#' @inheritParams pins::pin_write
+#' @param pins A list containing `board`, `name`, and `version`, as in
+#' [pins::pin_read()]
+#' @param display_pins Should the dashboard display a link to the pin(s)?
+#' Defaults to `TRUE`, but only creates a link if the pin contains a URL in
+#' its metadata.
+#' @param ... Arguments passed to [flexdashboard::flex_dashboard()]
+#'
+#' @details The `vetiver_dashboard()` function is a specialized type of
+#' \pkg{flexdashboard}. See the flexdashboard website for additional
+#' documentation:
+#'  \href{http://rmarkdown.rstudio.com/flexdashboard/}{http://rmarkdown.rstudio.com/flexdashboard/}
+#'
+#' Before knitting the example `vetiver_dashboard()` template, execute the
+#' helper function `pin_example_kc_housing_model()` to set up demonstration
+#' model and metrics pins needed for the monitoring demo. This function will:
+#'
+#' - fit an example model to training data
+#' - pin the vetiver model to your own [pins::board_local()]
+#' - compute metrics from testing data
+#' - pin these metrics to the same local board
+#'
+#' These are the steps you need to complete before setting up monitoring your
+#' real model.
+#'
+#' @export
+vetiver_dashboard <- function(pins, display_pins = TRUE, ...) {
+
+    rlang::check_installed("flexdashboard")
+    dashboard_dots <- rlang::list2(...)
+    v <- dashboard_read_version(
+        pins$board,
+        pins$name,
+        pins$version,
+        call = caller_env()
+    )
+
+    if (display_pins && !is_null(v$metadata$url)) {
+        dashboard_dots <- vetiver_modify_navbar(dashboard_dots, v$metadata$url)
+    }
+    if (is_null(dashboard_dots$favicon)) {
+        vetiver_fav <- system.file("favicon.ico", package = "vetiver")
+        dashboard_dots <- modifyList(dashboard_dots, list(favicon = vetiver_fav))
+    }
+
+    rmarkdown::output_format(
+        knitr = rmarkdown::knitr_options(
+            opts_knit = list(vetiver_dashboard.pins = pins),
+            # require to keep flexdashboard one
+            opts_chunk = list(),
+            knit_hooks = list(),
+            opts_hooks = list(),
+            opts_template = list()
+        ),
+        pandoc = NULL,
+        base_format = rlang::inject(flexdashboard::flex_dashboard(!!!dashboard_dots))
+    )
+}
+
+
+dashboard_read_version <- function(board, name, version, call) {
+    if (!pins::pin_exists(board, name)) {
+        rlang::abort(c(
+            glue("The pin {glue::double_quote(name)} does not exist on your board"),
+            "*" = "Check the `pins` params in your dashboard YAML",
+            "i" = glue("To knit the example vetiver monitoring dashboard, \\
+                       execute `vetiver::pin_example_kc_housing_model()` to \\
+                       set up demo model and metrics pins")
+        ))
+    }
```

Comment:
Nice way of handling the missing board! 

@juliasilge I am just wondering : You added a `call` argument, but you don't use it. Wasn't it to pass the `caller_env()` from `vetiver_dashboard()` into the call to `rlang::abort()` ? 

Maybe I misunderstood. Just sharing this feedback in case it was really the intention and it is just missing.

Reply:
Thank you so much for the heads up @cderv! I got distracted on this and left it unfinished yesterday but have fixed in #108.